tests/Microsoft.Identity.Test.Unit.net45/CoreTests/InstanceTests/InstanceDiscoveryManagerTests.cs

```diff
@@ -234,8 +253,46 @@ public async Task NetworkProviderIsCalledLastAsync()
 
             // Assert
             Assert.AreSame(_expectedResult, actualResult, "The known metadata provider should be queried second");
-            _knownMetadataProvider.Received(1).GetMetadata("some_env.com", otherEnvs);
-            _ = _networkMetadataProvider.Received(1).FetchAllDiscoveryMetadataAsync(authorityUri, _testRequestContext);
+            _knownMetadataProvider.Received(1).GetMetadata("some_env.com", otherEnvs, Arg.Any<ICoreLogger>());
+            _ = _networkMetadataProvider.Received(1).GetMetadataAsync(authorityUri, _testRequestContext);
+        }
+
+        [TestMethod]
+        public async Task UserProvider_TakesPrecedence_OverNetworkProvider_Async()
+        {
+            // Arrange
+            _discoveryManager = new InstanceDiscoveryManager(
+                _harness.HttpManager,
+                _harness.ServiceBundle.TelemetryManager,
+                false,
+                _userMetadataProvider,
+                _knownMetadataProvider,
+                _networkCacheMetadataProvider,
+                _networkMetadataProvider);
+
+            var otherEnvs = new[] { "env1", "env2" };
+            _userMetadataProvider.GetMetadataOrThrow("some_env.com", Arg.Any<ICoreLogger>()).Returns(_expectedResult);
+
+            // Act
+            InstanceDiscoveryMetadataEntry actualResult = await _discoveryManager.GetMetadataEntryTryAvoidNetworkAsync(
+                "https://some_env.com/tid",
+                otherEnvs,
+                _testRequestContext)
+                .ConfigureAwait(false);
+
+            InstanceDiscoveryMetadataEntry actualResult2 = await _discoveryManager.GetMetadataEntryAsync(
+               "https://some_env.com/tid",
+               _testRequestContext)
+               .ConfigureAwait(false);
+
+            // Assert
+            Assert.AreSame(_expectedResult, actualResult, "The user metadata provider should be queried second");
+            Assert.AreSame(_expectedResult, actualResult2, "The user metadata provider should be queried second");
+            _userMetadataProvider.Received(2).GetMetadataOrThrow("some_env.com", Arg.Any<ICoreLogger>());
+            _knownMetadataProvider.DidNotReceiveWithAnyArgs().GetMetadata(null, null, null);
+            _networkMetadataProvider.DidNotReceiveWithAnyArgs().GetMetadataAsync(
+                Arg.Any<Uri>(),
+                Arg.Any<RequestContext>()).GetAwaiter().GetResult();
```

Comment:
you are in an async method.  you should call await here instead of using GetAwaiter().GetResult()